Changelog for 0.1 release. (#304)
kubernetes-sigs/kube-agentic-networking: CHANGELOG/0.x-CHANGELOG.md

# v0.x Changelog

## Summary

## Table of Contents

- [v0.1.0](#v010)

# v0.1.0

This is the first release of the project. We introduce an `AccessPolicy` API to support agentic networking scenarios, starting with MCP-based authorization as well as core reference controller implementation for dynamic policy enforcement.

Initial release includes following:

- Security & Authentication Features

> - Access Control & API Extensions

Reviewer: ```suggestion
- XAccessPolicy API (Experimental)
```

> - MVP Controller

Reviewer: ```suggestion
- reference implementation
```

# v0.x Changelog

## Summary

> This inaugural release of the project introduces the core MVP controller implementation for dynamic policy enforcement and significantly extends the `AccessPolicy` API to support complex agentic networking scenarios, including identity federation and resource-pattern-based authorization.

Reviewer: nit: this is the first release so we dont really "extend AccessPolicy" but rather "introducing it".

Author: Changed to introduce.

> This inaugural release of the project introduces the core MVP controller implementation for dynamic policy enforcement and significantly extends the `AccessPolicy` API to support complex agentic networking scenarios, including identity federation and resource-pattern-based authorization.

Reviewer: ```suggestion
This is the first release of the project. We introduce an `AccessPolicy` API to support agentic networking scenarios, starting with MCP-based authorization as well as core reference controller implementation for dynamic policy enforcement.
```

Author: Done.

A major highlight is the introduction of secure, native mTLS authentication using SPIFFE identities (replacing JWTs) to support more flexible tool-gating models.

> The release also delivers critical improvements—such as enabling dual-stack IPv4/IPv6 resolution for external backends in the Envoy proxy—and greatly enhances the developer experience with automated quickstart container build pipelines, Hugging Face model integration for the ADK agent, and comprehensive E2E testing suites.

Reviewer: lets drop it not really necessary
```suggestion
```

Author: Done.

## Table of Contents

- [v0.1.0](#v010)

# v0.1.0

### Feature

- Adds a required field called `name` to AccessRule
  - Adds `maxItems: 10` kubebuilder marker to limit the length for both `rules` and `targetRefs` of the `AccessPolicy` resource (#58, @haiyanmeng)
- Extends the `AccessPolicy` API for more dynamic agentic scenarios such as: unbounded number of identities, identity federation, authorization based on common patterns of the targeted resources and/or authorization decision offloaded to external systems. (#17, @guicassolato)
- Handling resource deletion or modification of dependent object (#96, @HarshithaMS005)
- Implemented enqueueGatewaysForService with HTTPRoute backend matching (#84, @HarshithaMS005)
- Initial controller implementation. (#26, @chuangw6)
- Introduce standard gateway-name label for resource discovery and robust wait. (#103, @chuangw6)
- Support Hugging Face model in Quickstart ADK agent (#67, @chuangw6)
- Yes, MVP controller implementation (#28, @chuangw6)
- Yes, automate the the build and push processes of adk-agent and everything-mcp-server images used in quickstart. (#60, @chuangw6)

> - Yes, new API added (#20, @chuangw6)

Reviewer: what are these "Yes"es?

Author: idk, probably extracted from PR description. Removed them.

- 🚀 Feature: 
  - Implemented native mTLS authentication using SPIFFE identities issued by the internal KAN identity signer. 
    This replaces JWT-based identity with cryptographically secure certificates issued via Kubernetes 
    PodCertificateRequest resources.
  

>   ⚠️ User-Facing Changes: 

Reviewer: I dont think there are any user facing changes for this release, cause its the first one.

There will be starting next releases

Author: Moved them under `Feature`
